pkg/apis/workspaces/v1alpha1/commands.go

```diff
@@ -121,6 +121,23 @@ type ExecCommand struct {
 	// Optional list of environment variables that have to be set
 	// before running the command
 	Env []EnvVar `json:"env,omitempty"`
+
+	// +optional
+	// Whatever the command needs should be executed on background.
+	// If "false" the controller should wait for command to finish and report its status.
+	// If "false" the controller won't wait for it to finish (command can run indefinitely or until it is manually stopped).
+	// Default value is "true" for group.kind: run, debug.
+	// Default value is "false" for group.kind: build, test.
```

Comment:
I worry about setting different defaults depending on `group.kind`. Some builds, for example, are designed to pick up changes and rebuild automatically (e.g. with node). Having an optional value that defaults differently depending on `group.kind` means that the writer needs to cross-reference more fields in the spec to determine behaviour.

Reply:
for `restartOnChange` we could define one default value, but it will have to be `true`.

But I'm afraid that for `background` field we can't pick just one default value for all group kind. Whatever value we pick, it won't make sense for part of the commands :-( 
If we pick `false` it will make it  weird for `run` commands (especially in odo)
If we pick `true` it will make it really weird for most `build` commands :-(

Reply:
Yeah I don't see a good way around this issue. I'm fine with it as-is.